### PR TITLE
Added docstrings and type of parameters

### PR DESCRIPTION
**Pull request by SiddhantBobde**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/binary_tree/binary_search_tree.py [hunk 1]
```diff
@@ -141,6 +141,28 @@ def traversal_tree(self, traversal_function=None):
         else:
             return traversal_function(self.root)
 
+    def inorder(self, arr, Node):
```
Comment (cclauss) [inline data_structures/binary_tree/binary_search_tree.py:144]: Type hints are https://docs.python.org/3/library/typing.html

https://docs.python.org/3/library/doctest.html are also missing.

File: data_structures/binary_tree/binary_search_tree.py [hunk 2]
```diff
@@ -141,6 +141,28 @@ def traversal_tree(self, traversal_function=None):
...
+        Args:-
+        arr -> list
+        Node -> object of class Node
+        """
+        if Node:
+            self.inorder(arr, Node.left)
+            arr.append(Node.value)  # appends values of nodes to list arr
+            self.inorder(arr, Node.right)
+
+    def find_Kth_Smallest(self, k, Node):
```
Comment (cclauss) [inline data_structures/binary_tree/binary_search_tree.py:155]: Capital letters are not allowed in Python function names.

**Comment by cclauss** [comment]
Closing in favor of #2318

File: data_structures/binary_tree/binary_search_tree.py (hunk 1 above)
Comment (SiddhantBobde) [reply data_structures/binary_tree/binary_search_tree.py:144]: I am adding the type hints

**Pull request closed without merging**